hdfs-plugins/src/main/java/co/cask/hydrator/plugin/HDFSSink.java

```diff
@@ -76,7 +76,8 @@ public void prepareRun(BatchSinkContext context) throws Exception {
   public void transform(StructuredRecord input, Emitter<KeyValue<Text, NullWritable>> emitter) throws Exception {
     List<String> dataArray = new ArrayList<>();
     for (Schema.Field field : input.getSchema().getFields()) {
-      dataArray.add(input.get(field.getName()).toString());
+      String data = (input.get(field.getName()) != null) ? input.get(field.getName()).toString() : "";
+      dataArray.add(data);
```

Comment:
@anwar6953 I am with either "null" or "" (empty string). I don't think not adding is right because then the user has no idea which field went missing (say we have fname,lname,city and the lname is null, now if we don't put in anything then there is no way to know if the lname was null or city was null).